dags/hello_world/hello_world.py

```diff
@@ -0,0 +1,80 @@
+from datetime import timedelta
+
+from airflow import DAG
+from airflow.operators.python import PythonOperator
+from airflow.utils.dates import days_ago
+from airflow.models.taskinstance import TaskInstance
+from returns.curry import curry
+
+
+@curry
+def pull_execute_push(origin_key: str, target_key: str, function):
+    def wrapper(ti: TaskInstance):
+        data = ti.xcom_pull(key=origin_key)
+        ti.xcom_push(key=target_key, value=function(data))
+
+    return wrapper
+
+
+@curry
+def pull_execute(origin_key: str, function):
+    def wrapper(ti: TaskInstance):
+        data = ti.xcom_pull(key=origin_key)
+        function(data)
+
+    return wrapper
+
+
+@curry
+def execute_push(target_key: str, function):
+    def wrapper(ti: TaskInstance):
+        ti.xcom_push(key=target_key, value=function())
+
+    return wrapper
+
+
+def extract() -> str:
```

Comment:
As a pattern, I decided to write pure functions for tasks (except for `extract` and `load` for obvious reasons) because this isolates side effects and is therefore easy to test and robust to maintain and develop. These functions are then wrapped in the above decorators to be able to pass their values on and packaged in airflow's `PythonOperator` before stringed together in a DAG.